README.md

```diff
@@ -0,0 +1,123 @@
+
+### Services
+
+- **OrderService**: Manages the order processing logic, including placing orders in the database, retrieving order history, and calculating the total price of all items.
+- **BookService**: Manages the book processing logic, including book management for admins and book retrieval for users.
+- **CategoryService**: Manages category processing logic, including category management for admins and category retrieval for users.
+- **ShoppingCartService**: Handles the business logic related to shopping carts, including adding items and updating the cart.
+- **UserService**: Contains business logic for user operations such as registration and user retrieval by username (email).
+
+### Repositories
+
+- **BookRepository**: Interface for CRUD operations on Book entities and special methods for finding books by category.
+- **ShoppingCartRepository**: Interface for CRUD operations on ShoppingCart entities and methods for identifying a shopping cart by user ID.
+- **CartItemRepository**: Interface for CRUD operations on CartItem entities.
+- **OrderRepository**: Interface for CRUD operations on Order entities.
+- **OrderItemRepository**: Interface for CRUD operations on OrderItem entities and methods for retrieving order items by order ID.
+- **UserRepository**: Interface for CRUD operations on User entities and methods for checking if the user is already registered.
+- **RoleRepository**: Interface for CRUD operations on Role entities, including methods for identifying user roles.
+
+## Setup and Usage
+
+To get started with this project, follow these steps:
+
+1. **Clone the repository**:
+    ```sh
+    git clone https://github.com/bodya4243/online-book-store.git
+    cd online-book-store
+    ```
+
+2. **Build the project using Maven**:
+    ```sh
+    mvn clean install
+    ```
+
+3. **Run the application**:
+    ```sh
+    mvn spring-boot:run
+    ```
+
+4. **Access the application**:
+    - The application runs on `http://localhost:8080`.
+    - API documentation is available at `http://localhost:8080/swagger-ui.html`.
+
+5. **Swagger**:
+    - You can use Swagger for manipulating API endpoints.
+
+6. **Docker**:
+    - You can download the container from Docker and start the API.
+
+### Docker Setup
+
+1. **Build the Docker image**:
+    ```sh
+    docker build -t online-book-store .
+    ```
+
+2. **Run the Docker container**:
+    ```sh
+    docker run -d -p 8080:8080 online-book-store
+    ```
+
+3. **Verify the container is running**:
+    - Ensure the application is accessible at `http://localhost:8080`.
+
+## Postman Collection
+
+A collection of Postman requests has been created to help you test the APIs quickly. You can import the collection into Postman by following these steps:
+
+1. Download the Postman collection file from [link to your Postman collection].
+2. Open Postman and click on `Import`.
+3. Select the downloaded file and click `Import`.
```

Comment:
I do not think we need to download it, pretty sure postman can import a collection from URL